functional-tests/src/test/java/org/infinispan/online/service/endpoint/HotRodTester.java

```diff
@@ -41,11 +52,18 @@ public void testPutPerformance(URL urlToService, long timeout, TimeUnit timeUnit
       }
    }
 
-   private RemoteCache<String, String> getDefaultCache(URL urlToService) {
+   private RemoteCache<String, String> getDefaultCache(URL urlToService, boolean authenticate) {
       Configuration cachingServiceClientConfiguration = new ConfigurationBuilder()
          .addServer()
          .host(urlToService.getHost())
          .port(urlToService.getPort())
+         .security().authentication().enabled(authenticate)
```

Comment:
I think we could implement a decorator on `HotRodTester` called `AuthenticatedHotRodTested`. It could have different constructor (with username, password and all other required bits). This way it would be a bit cleaner and more extendable.

Reply:
I think the decorator pattern is overkill here and will require introducing unnecessary complexity with no real benefit. All of our services are going to use authorization by default, so the non-authenticated case is the exception and so a none authenticated client should only be used in the tests that ensure authorization is working as expected.

Reply:
That's a fair point actually. +1